include/vcpkg/base/system.proxy.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <optional>
+#include <string>
+
+/*
+ * This is a helper class. It reads IE Proxy settings.
+ * For coherence of vcpkg design "using HTTP(S)_PROXY not WPAD", it is a user-friendly design to manually
+ * read the IE Proxy settings on Windows, and set the environment variables HTTP(S)_PROXY to the same
+ * when starting an external program (like CMake, which file(DOWNLOAD) only accept proxy settings
+ * by HTTP(S)_PROXY variables).
+ */
+
+namespace vcpkg::System
+{
+    bool get_windows_ie_proxy_enabled();
+    std::optional<std::string> get_windows_ie_proxy_server();

```

Comment:
Please use Optional from base/ instead of std::optional; we don't get to assume C++17.

Reply:
OK.